### PR TITLE
RavenDB-12204: Document metadata won't have hasCounter/hasRevisions flags when migrating without counters/revisions. 

### PR DESCRIPTION
**Pull request by garayx**
Also metadata won't have counters array when migrating without counters.

### CODE REVIEW

File: src/Raven.Server/Documents/Document.cs [hunk 1]
```diff
@@ -70,6 +70,14 @@ public void EnsureMetadata()
             if (IndexScore.HasValue)
                 mutatedMetadata[Constants.Documents.Metadata.IndexScore] = IndexScore;
 
+            if (NonPersistentFlags.HasFlag(NonPersistentDocumentFlags.FromSmuggler) &&
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Document.cs]: This should be handled in StreamSource.GetDocuments.
We have a BlittableMetadataModifier there and you set a mode there that will strip the `@counters` property from the metadata when the document is being read.

**Review by ppekrol — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Document.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Server/Documents/Document.cs]: done

File: test/SlowTests/Issues/RavenDB-12204.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+using Xunit;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_12204 : RavenTestBase
+    {
+        [Fact]
+        public async Task CanMigrateFromRavenDb()
+        {
+            var file = Path.GetTempFileName();
```
Comment (ppekrol) [inline test/SlowTests/Issues/RavenDB-12204.cs]: use `NewDataPath` for this It will delete the directories/files when test will end.

Comment (garayx) [reply test/SlowTests/Issues/RavenDB-12204.cs]: done

**Review by ppekrol — APPROVED** [review]

**Review by ayende — APPROVED** [review]

**Pull request merged**